spotbugs/src/main/java/edu/umd/cs/findbugs/util/ClassName.java

```diff
@@ -175,9 +175,47 @@ public static String extractSimpleName(@DottedClassName String className) {
      * @return true if it's a valid class name, false otherwise
      */
     public static boolean isValidClassName(String className) {
-        // FIXME: should use a regex
+        return isValidBinaryClassName(className) ||
+                isValidDottedClassName(className) ||
+                isValidArrayFieldDescriptor(className) ||
+                isValidClassFieldDescriptor(className);
+    }
+
+    private static boolean isValidBinaryClassName(String className) {
+        return className.indexOf('.') == -1 &&
+                className.indexOf('[') == -1 &&
+                className.indexOf(';') == -1;
```

Comment:
According to the [JVMS 4.2.2](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.2.2):

> An unqualified name must contain at least one Unicode code point and must not contain any of the ASCII characters . ; [ / (that is, period or semicolon or left square bracket or forward slash).

So better to check existence of `/` too.

Reply:
I don't think this is entirely correct. An unqualified name may not contain `.`, `;`, `[`, or `/`, as you've quoted, but a binary class name consists of one or more unqualified names, which are separated by `/`s.

According to [JVMS 4.2.1](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.2.1): 

>  In this internal form, the ASCII periods (.) that normally separate the identifiers which make up the binary name are replaced by ASCII forward slashes (/). The identifiers themselves must be unqualified names (§4.2.2).

If we were to exclude `/` here, names of the form `com/foo/Example` would be rejected.

I suggest that this remains as is, but I am happy to take any further suggestions.